src/Titon/Console/Command/AbstractCommand.hh

```diff
@@ -0,0 +1,344 @@
+     * The `Input` object containing all registered and parsed command line
+     * parameters.
+     *
+     * @var \Titon\Console\Input
+     */
+    protected Input $input;
+
+    /**
+     * The name of the command passed into the command line.
+     *
+     * @var string
+     */
+    protected string $name = '';
+
+    /**
+     * Bag container holding all registered `Option` objects
+     *
+     * @var \Titon\Console\InputBag<Option>
+     */
+    protected InputBag<Option> $options;
+
+    /**
+     * The `Output` object to handle output to the user.
+     *
+     * @var \Titon\Console\Output
+     */
+    protected Output $output;
+
+    /**
+     * Construct a new instance of a command.
+     */
+    public function __construct(Input $input, Output $output) {
+        $this->input = $input;
+        $this->output = $output;
+        $this->arguments = new InputBag();
+        $this->flags = new InputBag();
+        $this->options = new InputBag();
+    }
+
+    /**
+     * Add a new `Argument` to be registered and parsed with the `Input`.
+     *
+     * @param \Titon\Console\InputDefinition\Argument $argument
+     *
+     * @return $this
+     */
+    public function addArgument(Argument $argument): this {
+        $this->arguments->set($argument->getName(), $argument);
+
+        return $this;
+    }
+
+    /**
+     * Add a new `Flag` to be registered and parsed with the `Input`.
+     *
+     * @param \Titon\Console\InputDefinition\Flag $flag
+     *
+     * @return $this
+     */
+    public function addFlag(Flag $flag): this {
+        $this->flags->set($flag->getName(), $flag);
+
+        return $this;
+    }
+
+    /**
+     * Add a new `Option` to be registered and parsed with the `Input`.
+     *
+     * @param \Titon\Console\InputDefinition\Option $option
+     *
+     * @return $this
+     */
+    public function addOption(Option $option): this {
+        $this->options->set($option->getName(), $option);
+
+        return $this;
+    }
+
+    /**
+     * Construct and return a new `Menu` object given the choices and display
+     * message.
+     *
+     * @param Map<mixed, mixed> $choices    Accepted values
+     * @param string $message               The message to display before the choices
+     *
+     * @return \Titon\Console\UserInput\Menu
+     */
+    protected function confirm(Map<mixed, mixed> $choices, string $message = ''): Menu {
+        $menu = new Menu($this->input, $this->output);
+        $menu->setAcceptedValues($choices)->setMessage($message);
+
+        return $menu;
+    }
+
+    /**
+     * Alias method for sending output through STDERROR.
+     *
+     * @param string $output    The message to send
+     */
+    protected function error(string $output): void {
+        $this->output->error($output);
+    }
+
+    /**
+     * Retrieve an `Argument` value by key.
+     *
+     * @param string $key
+     *
+     * @return mixed
+     */
+    protected function getArgument(string $key): ?string {
```

Comment:
These are easy aliases to the Input value getter of the definitions. I'm pushing up a commit that allows a default value to be  passed in, so null could be valid here.